src/object/cli_obj.c

```diff
@@ -2216,6 +2216,18 @@ obj_iod_sgl_valid(daos_obj_id_t oid, unsigned int nr, daos_iod_t *iods,
 				return -DER_INVAL;
 			}
 		}
+		if (sgls != NULL && sgls[i].sg_nr > 0) {
+			d_sg_list_t	*sg = &sgls[i];
+			d_iov_t		*iov;
+
+			for (j = 0; j < sg->sg_nr; j++) {
+				iov = sg->sg_iovs + j;
+				if (iov == NULL || (iov->iov_buf_len > 0 && iov->iov_buf == NULL)) {
+					D_ERROR("Bad iov\n");
```

Comment:
it's better to make the error message more informative, so that one can tell which IOV is bad and why it's bad IOV.

Reply:
OK, I can refine the log in following PR, thx

Reply:
since a re-run is needed with Features since DFS was changed, please change it with this PR.